train.py

```diff
@@ -422,23 +462,36 @@ def train(hyp, opt, device, tb_writer=None):
         logger.info('%g epochs completed in %.3f hours.\n' % (epoch - start_epoch + 1, (time.time() - t0) / 3600))
         if opt.data.endswith('coco.yaml') and nc == 80:  # if COCO
             for m in (last, best) if best.exists() else (last):  # speed, mAP tests
+                test_model = attempt_load(m, device) if not qat else model
                 results, _, _ = test.test(opt.data,
                                           batch_size=batch_size * 2,
                                           imgsz=imgsz_test,
                                           conf_thres=0.001,
                                           iou_thres=0.7,
-                                          model=attempt_load(m, device).half(),
+                                          model=test_model.half() if not opt.disable_amp else test_model,
                                           single_cls=opt.single_cls,
                                           dataloader=testloader,
                                           save_dir=save_dir,
                                           save_json=True,
                                           plots=False,
-                                          is_coco=is_coco)
+                                          is_coco=is_coco,
+                                          half_precision=not opt.disable_amp)
+
+        # ONNX export
+        if opt.export_onnx:
+            onnx_path = f'{save_dir}/model.onnx'
+            logger.info(f'training complete, exporting ONNX to {onnx_path}')
+            export_model = model.module if is_parallel_model(model) else model
+            export_model.model[-1].export = True  # do not export grid post-procesing
+            exporter = ModuleExporter(export_model, save_dir)
+            exporter.export_onnx(torch.randn(1, 3, imgsz, imgsz), convert_qat=True)
+            if qat:
+                skip_onnx_input_quantize(onnx_path, onnx_path)
```

Comment:
what is `skip_onnx_input_quantize` doing and why does it have the same `onnx_path` arg twice?

Reply:
this is the optimization pass to remove the initial `QuantizeLinear` node in an ONNX graph so the model can take uint8 inputs.  the first two arguments are the model file and the save file.  gave the same file for both so it will override the original.